src/plugins/discover/public/application/main/hooks/use_saved_search_messages.ts

```diff
@@ -37,15 +38,32 @@ export function sendCompleteMsg(main$: DataMain$, foundDocuments = true) {
   if (main$.getValue().fetchStatus === FetchStatus.COMPLETE) {
     return;
   }
-  main$.next({ fetchStatus: FetchStatus.COMPLETE, foundDocuments, error: undefined });
+  main$.next({
+    ...main$.getValue(),
+    fetchStatus: FetchStatus.COMPLETE,
+    foundDocuments,
+    error: undefined,
+  });
+}
+
+/**
+ * Send message containing timeRange via main observable
+ */
+export function sendTimeRangeMsg(
+  main$: DataMain$,
+  timeRange: TimeRange,
+  timeRangeRelative?: TimeRange,
+  query?: AggregateQuery | Query | undefined
```

Comment:
this could be the the "setQueryStartMsg" msg, store all query  information as source of truth for dependend operations in the UI .. or this info could be stored in internal state, for sure, ... wherever, one source of truth